openmc/element.py

```diff
@@ -102,9 +123,160 @@ def expand(self):
+            absent_nuclides = natural_nuclides.difference(mutual_nuclides)
+            mutual_nuclides = sorted(list(mutual_nuclides))
+            absent_nuclides = sorted(list(absent_nuclides))
+
+            # If all natural nuclides are present in the library, expand element
+            # using all natural nuclides
+            if len(absent_nuclides) == 0:
+                for nuclide, abundance in sorted(NATURAL_ABUNDANCE.items()):
+                    if re.match(r'{}\d+'.format(self.name), nuclide):
+                        nuc = openmc.Nuclide(nuclide)
+                        nuc.scattering = self.scattering
+                        nuclides.append(nuc)
+                        abundances.append(abundance)
+
+            # If no natural elements are present in the library, check if the
+            # 0 element is present. If so, set the abundance to 1 for this
+            # nuclide. Else, raise an error.
+            elif len(mutual_nuclides) == 0:
+                if 0 in library_nuclides:
+                    nuc = openmc.Nuclide(self.name + '0')
+                    nuc.scattering = self.scattering
+                    nuclides.append(nuc)
+                    abundances.append(1.0)
+                else:
+                    msg = 'Unable to expand element {0} because the cross '\
+                          'section library provided does not contain any of '\
+                          'the natural isotopes for that element.'\
+                          .format(self.name)
+                    raise ValueError(msg)
+
+            # If some, but not all, natural nuclides are in the library, add
+            # the mutual nuclides. For the absent nuclides, increment the
+            # abundance of the nearest mutual nuclide with their abundances.
+            else:
+
+                # Add the mutual isotopes
+                nuclides_a = []
+                for nuclide, abundance in sorted(NATURAL_ABUNDANCE.items()):
+                    if re.match(r'{}\d+'.format(self.name), nuclide) and \
+                       int(nuclide[name_len:]) in mutual_nuclides:
+                        nuc = openmc.Nuclide(nuclide)
+                        nuc.scattering = self.scattering
+                        nuclides.append(nuc)
+                        nuclides_a.append(int(nuclide[name_len:]))
+                        abundances.append(abundance)
+
+                # Adjust the abundances for the absent nuclides
+                for nuclide, abundance in sorted(NATURAL_ABUNDANCE.items()):
+                    if re.match(r'{}\d+'.format(self.name), nuclide) and \
+                       int(nuclide[name_len:]) in absent_nuclides:
+
+                        # Get index to the nearest nuclide
+                        a = int(nuclide[name_len:])
+                        i = min(list(range(len(nuclides_a))), key=lambda j: \
+                                abs(nuclides_a[j] - a))
```

Comment:
The elements and associated isotopes of concern are:

**Some isotopes lumped together**
- **O**: Some libraries lump O-18 with O-16 and some lump O-18 and O-17 with O-16.
- **Ta**: Some libraries lump Ta-180 together with Ta-181.
- **W**: Some libraries lump W-180 with W-182, but keep in mind W-181 is not a natural isotope.

**All isotopes lumped together**
- **V**: Some libraries lump V-50 and V-51 as V-0.
- **C**: All libraries lump C-12 and C-13 together as C-0.
-  **Zn**: Some libraries lump all the isotopes together as Zn-0.
- **Ga**: Some libraries lump Ga-69 and Ga-71 together as Ga-0.
- **Os**: Some libraries lump all the isotopes together as Os-0.
- **Pt**: Some libraries lump all the isotopes together as Pt-0.
- **Tl**: Some libraries lump all the isotopes together as Tl-0.

I didn't realize O-18 was lumped with O-16 instead of O-17. I'm just going to break up the "some isotopes lumped together" cases individually and print an error if it is a case other than those.